irs/core/upload_handler.py

```diff
@@ -0,0 +1,235 @@
+    async with KafkaEventPublisher.construct(config=CONFIG) as publisher:
+        if total_checksum == checksum:
+            topic = "upload_validation_success"
+            data = UploadValidationSuccessEvent(
+                content_id=total_checksum,
+                secret_id=secret_id,
+                offset=offset,
+                part_size=PART_SIZE,
+                part_checksums_md5=part_checksums[0],
+                part_checksums_sha256=part_checksums[1],
+            ).dict()
+        else:
+            topic = "upload_validation_failure"
+            # implement exception handling, get cause from exception
+            data = UploadValidationFailureEvent(
+                file_id=object_id, cause="Checksum mismatch"
+            ).dict()
+        publisher.publish(paylod=data, type_=type_, key=key, topic=topic)
+
+
+async def get_download_url(*, object_id: str):
+    """Get object download URL frome"""
+    config = S3Config(
+        s3_endpoint_url=CONFIG.s3_endpoint_url,
+        s3_access_key_id=CONFIG.s3_access_key_id,
+        s3_secret_access_key=CONFIG.s3_secret_access_key,
+    )
+    storage = S3ObjectStorage(config=config)
+
+    return storage.get_object_download_url(
+        bucket_id=CONFIG.bucket_id, object_id=object_id
+    )
+
+
+async def send_to_eks(
+    *, file_part: bytes, public_key: str, eks_url: str
+) -> Tuple[bytes, str, int]:
+    """Get encryption secret and file content offset from envelope"""
+    data = base64.b64encode(file_part).hex()
+    request_body = {"public_key": public_key, "file_part": data}
+    try:
+        response = requests.post(url=eks_url, json=request_body, timeout=60)
+    except requests.exceptions.RequestException as request_error:
+        raise exceptions.RequestFailedError(url=eks_url) from request_error
+
+    status_code = response.status_code
+    # implement httpyexpect error conversion
+    if status_code != 200:
+        spec = {
+            400: {
+                "malformedOrMissingEnvelopeError": exceptions.MalformedOrMissingEnvelope()
+            },
+            403: {"envelopeDecryptionError": exceptions.EnvelopeDecryptionError()},
+        }
+        ResponseExceptionTranslator(spec=spec).handle(response=response)
+        raise exceptions.BadResponseCodeError(url=eks_url, response_code=status_code)
+
+    body = response.json()
+    secret = base64.b64decode(codecs.decode(body["secret"], "hex"))
+    secret_id = body["secret_id"]
+    offset = body["offset"]
+
+    return secret, secret_id, offset
+
+
+async def compute_checksums(
+    *, download_url: str, secret: bytes, object_size: int, offset: int
+) -> Tuple[Tuple[List[str], List[str]], str]:
+    """Compute total unencrypted file checksum and encrypted part checksums"""
+    total_sha256_checksum = hashlib.sha256()
+    encrypted_md5_part_checksums = []
+    encrypted_sha256_part_checksums = []
+
+    # buffers to account for non part/cipher segment size aligned blocks
+    partial_chunk = b""
```

Comment:
The term chunk could be easily confused with part. May rather use the term cyper_segment. E.g.:
```suggestion
    partial_cypher_segment = b""
```